packages/backend/src/modules/interop/engine/financials/TokenDb.ts

```diff
@@ -1,86 +1,68 @@
-import type { Branded } from '@l2beat/shared-pure'
-import mockData from './TokenDbMockData.json'
+import type { Logger } from '@l2beat/backend-tools'
+import type { TokenDbClient } from '@l2beat/token-backend'
+import { DeployedTokenId } from './DeployedTokenId'
 
-export type AbstractTokenId = Branded<string, 'AbstractTokenId'>
-
-export function AbstractTokenId(id: string) {
-  return id as AbstractTokenId
-}
-
-AbstractTokenId.from = function from(
-  realId: string,
-  issuer: string,
-  symbol: string,
-) {
-  return `${realId}:${issuer}:${symbol}`
-}
-
-AbstractTokenId.realId = function readId(id: AbstractTokenId) {
-  return id.slice(0, 6)
-}
-
-AbstractTokenId.issuer = function issuer(id: AbstractTokenId) {
-  const withoutRealId = id.slice(7)
-  return withoutRealId.slice(0, withoutRealId.indexOf(':'))
-}
-
-AbstractTokenId.symbol = function symbol(id: AbstractTokenId) {
-  const withoutRealId = id.slice(7)
-  return withoutRealId.slice(withoutRealId.indexOf(':') + 1)
-}
-
-export type DeployedTokenId = Branded<string, 'AbstractTokenId'>
-
-export function DeployedTokenId(id: string) {
-  return id as DeployedTokenId
-}
-
-DeployedTokenId.from = function from(chain: string, address: string) {
-  return `${chain}+${address}` as DeployedTokenId
-}
-
-DeployedTokenId.chain = function chain(id: DeployedTokenId) {
-  return id.slice(0, id.indexOf('+'))
-}
-
-DeployedTokenId.address = function address(id: DeployedTokenId) {
-  return id.slice(id.indexOf('+') + 1)
-}
-
-export type PriceInfo = Map<
+export type TokenInfos = Map<
   DeployedTokenId,
   {
-    abstractId: AbstractTokenId
+    abstractId: string
+    symbol: string
     decimals: number
     coingeckoId: string
   }
 >
 
-export interface ITokenDb {
-  getPriceInfo(deployedTokens: DeployedTokenId[]): Promise<PriceInfo>
-}
+export class TokenDb {
```

Comment:
IMO this entire class should go and the InteropFinancials should receive the client directly